src/legacy/plugin_discovery/types.ts

```diff
@@ -72,8 +72,10 @@ export interface LegacyPluginOptions {
     visTypes: string[];
     embeddableActions?: string[];
     embeddableFactories?: string[];
+    interpreter?: string[];
+    uiSettingDefaults?: Record<string, any>;
   }>;
-  uiCapabilities?: Capabilities;
+  uiCapabilities?: Capabilities | (() => Partial<Capabilities>);
```

Comment:
@lukeelmers  In Timelion uiCapabilities is a funciton. I offer @Avinar-24  to use object instead. Are you agree or we should modify this interface?  

It will be something like: 
![image](https://user-images.githubusercontent.com/20072247/64349217-3982e300-cfff-11e9-8f80-faad997df6bc.png)


I don't like idea to modify interface like here cause in future in all places you should check is uiCapabilities object or it's a function. We should avoid of this practice

Reply:
If we dive deeper in the situation we can see that `Capabilities` is a sustainable interface with 3 necessary fields (`navLinks, management, catalogue`) which doesn't need to be implemented in `uiCapabilities` of the `Timelion`.

If we make the fields as optional inside the `Capabilities` we should recheck 30 places where it's used. It may cost much more than the written solution.

Reply:
Moreover the `|` symbol is used in many places and it doesn't seem the situation differs from others.

Reply:
I checked with the security team who built the `uiCapabilities` feature. Turns out the type in this file is actually incorrect and should be as follows:
```ts
uiCapabilities?: () => Capabilities | Promise<Capabilities>;
```

Will that work?

Reply:
No, it won't. In that case we also need to implement unnecessary 3 fields: `(navLinks, management, catalogue)`. But `(() => Partial<Capabilities>) | Promise<Capabilities>;` will work fine.

Reply:
@lukeelmers Should we change `uiCapabilities` that way? ^^

Reply:
What's an option is better?
1. `(() => Partial<Capabilities>) | Capabilities`
2. `(() => Partial<Capabilities>) | Promise<Capabilities>;`

Reply:
Left the first one.

Reply:
^^ @ppisljar

Reply:
`Partial<Capabilities>) | Promise<Partial<Capabilities>>;`

@legrego do you agree ?

Reply:
Sorry, I'm having a hard time following this thread. Is the main issue that we don't want to have to specify `navLinks`, `management`, and `catalogue` for the Timelion plugin?

> In Timelion uiCapabilities is a funciton. I offer @Avinar-24 to use object instead. Are you agree or we should modify this interface?

I think it makes sense to keep this as a function, rather than changing this to an object. Capabilities haven't been ported to NP yet, so if we're going to make a change to the interface, I think it'd make sense to do so at that time.

Reply:
@legrego It seems not easy to do. If we'd like to keep `uiCapabilities()` as a function it's needed to tweak the field a little inside the `LegacyPluginOptions`interface. I've left a comment over the field. If it suits you, please approve the PR due to last thread.

Reply:
I think @ppisljar's suggestion of `Partial<Capabilities>) | Promise<Partial<Capabilities>>` is the way to go here

Reply:
@legrego, @ppisljar it would work if `uiCapabilities` was an object or an async function.

We can temporary ignore the check by leaving a link to the discussion. And when `Timelion` is ported to NP with other plugins which use `Capabilities` we will revise it again